pkg/v1/tkg/client/machine_deployment_utkg.go

```diff
@@ -0,0 +1,284 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	capi "sigs.k8s.io/cluster-api/api/v1beta1"
+
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/clusterclient"
+)
+
+func DoSetMachineDeploymentCC(clusterClient clusterclient.Client, cluster *capi.Cluster, options *SetMachineDeploymentOptions) error {
```

Comment:
This function is too long (almost 200 lines!). There's little chance we'll be able reason about it in a short time.

Can we structure it so that it is clear from the lower level function names (that it's going to call) what exactly it intends to do? Ideally, minimizing the amount of conditional logic.

These lower level functions should also be structured in similar way.

I see a few candidates to encapsulate in their own functions, like populating cluster variables: `nodeLabelsVar`, `workerVariable`, `vcenterVariable`.